In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
data=pd.read_excel('C:/Users/Vishal/Desktop/LU aiml/Data/Linear Regression.xlsx')
data.head()

price  sqft_living  bedrooms  bathrooms  floors
0  221900         1180         3       1.00     1.0
1  538000         2570         3       2.25     2.0
2  180000          770         2       1.00     1.0
3  604000         1960         4       3.00     1.0
4  510000         1680         3       2.00     1.0

# Linear regression model of price vs Bathrooms

In [3]:
linbath=data.drop(['sqft_living','bedrooms','floors'],axis=1)
linbath.head()

price  bathrooms
0  221900       1.00
1  538000       2.25
2  180000       1.00
3  604000       3.00
4  510000       2.00

In [4]:
x=linbath[['bathrooms']]
y=linbath[['price']]

In [11]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.25)


In [12]:
from sklearn.linear_model import LinearRegression
linreg=LinearRegression()

In [13]:
linreg.fit(x_train,y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [17]:
linreg.coef_

array([[249565.50865919]])

In [18]:
linreg.intercept_

array([11179.93030721])

__Equation is:__

__price = 13513.13889573 + 249375.03130974 (bathrooms)__

In [14]:
ypred=linreg.predict(x_test)

In [15]:
from sklearn.metrics import r2_score, mean_squared_error
r2_score(y_test,ypred)

0.28820378817108283

In [16]:
mean_squared_error(y_test,ypred)

94305042951.49368

# INFERENCE:
R-square value is 0.28820378817108283 means there is 28.8% variance in the target variable.

Totat Error in the model is 94305042951.4936